extensions/bustache/ApplyTemplate.cpp

```diff
@@ -19,81 +19,76 @@
+  {}
+  int64_t process(const std::shared_ptr<io::BaseStream>& stream) override {
+    logger_->log_info("ApplyTemplate reading template file from %s", template_file_);
+    // TODO(szaszm): we might want to return to memory-mapped input files when the next todo is done. Until then, the agents stores the whole result in memory anyway, so no point in not doing the same
+    // with the template file itself
+    const auto template_file_contents = [this] {
+      std::ifstream ifs{template_file_};
+      return std::string{std::istreambuf_iterator<char>{ifs}, std::istreambuf_iterator<char>{}};
+    }();
+
+    bustache::format format(template_file_contents);
+    bustache::object data;
+
+    for (const auto &attr : flow_file_.getAttributes()) {
+      data[attr.first] = attr.second;
+    }
+
+    // TODO(calebj) write ostream reciever for format() to prevent excessive copying
+    std::string ostring = to_string(format(data));
+    stream->write(gsl::make_span(ostring).as_span<const std::byte>());
+    return gsl::narrow<int64_t>(ostring.length());
+  }
 
-namespace org {
-namespace apache {
-namespace nifi {
-namespace minifi {
-namespace processors {
-
-core::Property ApplyTemplate::Template("Template", "Path to the input mustache template file", "");
-core::Relationship ApplyTemplate::Success("success", "success operational on the flow record");
+ private:
+  std::shared_ptr<core::logging::Logger> logger_ = core::logging::LoggerFactory<ApplyTemplate>::getLogger();
+  std::filesystem::path template_file_;
+  const core::FlowFile& flow_file_;
+};
+}  // namespace
 
 void ApplyTemplate::initialize() {
-  //! Set the supported properties
-  std::set<core::Property> properties;
-  properties.insert(Template);
-  setSupportedProperties(properties);
-  //! Set the supported relationships
-  std::set<core::Relationship> relationships;
-  relationships.insert(Success);
-  setSupportedRelationships(relationships);
+  setSupportedProperties({Template});
+  setSupportedRelationships({Success});
 }
 
 void ApplyTemplate::onTrigger(const std::shared_ptr<core::ProcessContext> &context,
                               const std::shared_ptr<core::ProcessSession> &session) {
   auto flow_file = session->get();
-
   if (!flow_file) {
     return;
   }
 
   std::string template_file;
   context->getProperty(Template, template_file, flow_file);
-  WriteCallback cb(template_file, flow_file);
+  WriteCallback cb(template_file, *flow_file);
   session->write(flow_file, &cb);
   session->transfer(flow_file, Success);
 }
 
-ApplyTemplate::WriteCallback::WriteCallback(const std::string &path, const std::shared_ptr<core::FlowFile> &flow_file) {
-  template_file_ = path;
-  flow_file_ = flow_file;
-}
-
-int64_t ApplyTemplate::WriteCallback::process(const std::shared_ptr<io::BaseStream>& stream) {
-  logger_->log_info("ApplyTemplate reading template file from %s", template_file_);
-  boost::iostreams::mapped_file_source file(template_file_);
```

Comment:
This line was crashing (segfault) when compiled with libc++, so I removed it. I also saw a bunch of refactoring opportunities, so ApplyTemplate has quite a few changes. As it turns out, this line was the only reason the bustache extension required boost, so I removed that requirement and checked a few boost-related pieces of code.